drivers/include/n25q128.h

```diff
@@ -0,0 +1,206 @@
+ *          parameters to the configuration type (n25q128_conf_t). The other
+ *          structures must not be modified.
+ *
+ * @param[in] *dev  A pointer to the configured device.
+ */
+int n25q128_init(n25q128_dev_t *dev);
+
+/*
+ * @brief   Reads the content of the "Read Identification"-register.
+ *
+ * The Read Identification (RDID) instruction allows to read the device
+ * identification data:
+ * > Manufacturer identification (1 byte)
+ * > Device identification (2 bytes)
+ * > A Unique ID code (UID) (17 bytes, of which 14 factory programmed upon
+ * customer request).
+ *
+ * @param[in] *dev      A pointer to the configured device.
+ * @param[out] *buf     Stores the id from the memory device.
+ * @param[in] buf_len   The length of the *buf.
+ */
+void n25q128_read_id(n25q128_dev_t *dev, uint8_t *buf, size_t len);
+
+/*
+ * @brief   Read a specific amount of data bytes from an address.
+ *
+ * @param[in] *dev      A pointer to the configured device.
+ * @param[in] address   The address to read bytes from.
+ * @param[out] *buf     Stores the bytes from the address of the device.
+ * @param[in] buf_len   The length of the *buf.
+ */
+void n25q128_read_data_bytes(n25q128_dev_t *dev, uint32_t addr, uint8_t *buf, size_t len);
+
+/*
+ * @brief   Programs bytes into the memory.
+ *
+ * The Page Program (PP) instruction allows bytes to be programmed in the memory
+ * (changing bits from 1 to 0). Before it can be accepted, a Write Enable (WREN)
+ * instruction must previously have been executed. After the Write Enable (WREN)
+ * instruction has been decoded, the device sets the Write Enable Latch (WEL).
+ *
+ * @param[in] *dev      A pointer to the configured device.
+ * @param[in] address   The address to read bytes from.
+ * @param[out] *buf     Stores the bytes from the address of the device.
+ * @param[in] buf_len   The length of the *buf.
+ */
+void n25q128_page_program(n25q128_dev_t *dev, uint32_t addr, uint8_t *buf, size_t len);
+
+/*
+ * @brief   Erases all bits (set to 1 (FFh)) inside the chosen sector.
+ *
+ * NOTE: This needs to be done, before programming a sector with new values.
+ *       Because the memory can only set bits back to 1 during erasing. So, once
+ *       a bit is programmed from 1 to 0, it cannot be 'overwritten'.
+ *
+ * @param[in] *dev      A pointer to the configured device.
+ * @param[in] addr      The address of the sector to erase.
+ */
+void n25q128_sector_erase(n25q128_dev_t *dev, uint32_t addr);
+
+/*
+ * @brief   Erases the whole device.
+ *
+ * The Bulk Erase (BE) instruction sets all bits to '1' (FFh). Before it can be
+ * accepted, a Write Enable (WREN) instruction must previously have been
+ * executed. After the Write Enable (WREN) instruction has been decoded, the
+ * device sets the Write Enable Latch (WEL).
+ */
+void n25q128_bulk_erase(n25q128_dev_t *dev);
+
+/*
+ * @brief   Pause the program/erase program on the memory.
+ *
+ */
+void n25q128_program_erase_suspend(n25q128_dev_t *dev);
```

Comment:
Does the chip remember the position? Will it return busy if someone suspended a mass erase and tries to write data immediately after?